internal/scan-manager/core/src/main/java/org/wso2/security/tools/scanmanager/core/service/ScanEngineServiceImpl.java

```diff
@@ -266,47 +263,126 @@ private void initiateScanRequest(Scan scan, ScannerApp scannerApp) throws ScanMa
+     * will be restarted.
+     *
+     * @param schduler ScheduledExecutorService
+     */
+    private void activateScanResilienceTask(ScheduledExecutorService schduler) {
+        Runnable checkContainerStatusTask = new CheckContainerStatusTask();
+        schduler.scheduleWithFixedDelay(checkContainerStatusTask, INITIAL_DELAY, DELAY_BETWEEN_RUNS, TimeUnit.MINUTES);
+    }
+
+    /**
+     * Perform scan recovery task upon availability of container.
+     *
+     * @param containerInfo Container information
+     * @param scan          Scan object
+     */
+    private void doScanRecovery(Container containerInfo, Scan scan) {
+
+        // If container is not exist, restart the container
+        try {
+            dockerHandler.restart(containerInfo.getId());
+
+            // Sleep till the scanner service is started.
+            Thread.sleep(SCANNER_SERVICE_WAIT_TIME);
+        } catch (ScanManagerException | InterruptedException e) {
+            logService.insert(scan, LogType.ERROR,
+                    "Error occurred while restarting the container : " + containerInfo.getId());
+        }
+        logService.insert(scan, LogType.INFO, "Scanner container restarted. Container id: " + scan.getContainerId());
+        ScannerApp scannerApp = scannerService.getByScannerAndAppId(scan.getScanner(), scan.getScannerAppId());
+
+        // Initiate the scan request
+        try {
+            sendStartScanRequest(containerInfo, scannerApp, scan, true);
+            logService.insert(scan, LogType.INFO, "Scan is resumed for job Id : " + scan.getJobId());
+        } catch (ScanManagerException e) {
+            logService.insert(scan, LogType.ERROR, "Error occured while resuming scan : " + containerInfo.getId());
+        }
+    }
+
+    private final class CheckContainerStatusTask implements Runnable {
+
+        @Override
+        public void run() {
+            List<Scan> listofActiveScans;
+            listofActiveScans = scanService.getByStatus(ScanStatus.RUNNING);
+            listofActiveScans.addAll(scanService.getByStatus(ScanStatus.SUBMITTED));
+
+            for (Scan scan : listofActiveScans) {
+
+                // Get the container Id of running or submitted scan.
+                Container containerInfo = null;
+                try {
+                    containerInfo = dockerHandler.inspect(scan.getContainerId());
+                } catch (ScanManagerException e) {
+                    logService.insert(scan, LogType.ERROR,
+                            "Error occured while getting the container information : " + scan.getJobId());
+                }
+
+                // Here Container from life cycle "STOP" and "DESTROY" which is not casued by manual interaction,
+                // needs to be restarted. If container is in that state, container needs to be restarted and scan
+                // needs to be resumed.
+                if (containerInfo != null) {
```

Comment:
Better to move these also into the try block. If `doScanRecovery`  is not expected to throw an ScanManagerException.

Reply:
resolved